core/sql/generator/GenExplain.cpp

```diff
@@ -660,6 +660,269 @@ FileScan::addSpecificExplainInfo(ExplainTupleMaster *explainTuple,
   return(explainTuple);
 }
 
+static void appendListOfColumns(Queue* listOfColNames,ComTdb *tdb, NAString& outNAString){
+
+	if (((ComTdbHbaseAccess*)tdb)->sqHbaseTable()){// if trafodion table
+	  char buf[1000];
+
+	  listOfColNames->position();
+	  for (Lng32 j = 0; j < listOfColNames->numEntries(); j++)
+	    {
+	      char * currPtr = (char*)listOfColNames->getCurr();
+
+	      Lng32 currPos = 0;
+	      Lng32 jj = 0;
+	      short colNameLen = *(short*)currPtr;
+	      currPos += sizeof(short);
+	      char colFam[100];
+	      while (currPtr[currPos] != ':')
+		{
+		  colFam[jj] = currPtr[currPos];
+		  currPos++;
+		  jj++;
+		}
```

Comment:
code compaction vs readability is a long debated topic and end up being developer taste specific. I personally prefer compaction too, but wanted to respect the model I followed to favor consistancy over my personal taste.  see the code I refer too:
  for (Lng32 i = 0; i < colNameLen; i++)
    {
      colName[i] = currPtr[currPos];
      currPos++;
    }